### PR TITLE
Fix for #799

### PR DESCRIPTION
**Pull request by apdavison**
Adds t_start and related properties to ImageSequence; 

Also renames "sampling_period" to "frame_duration", as I think this is more meaningful. As ImageSequence was added only recently, I decided against adding a deprecation period for this change.

Note that the important changes are in the first commit; the second commit is just formatting and editing of language.

### CODE REVIEW

**Comment by apdavison** [comment]
Fixes for NixIO needed...

**Comment by achilleas-k** [comment]
I think most of the issues with the failing tests are things that try to be general solutions to testing but fall on their face when things change in any meaningful way.  I'd like to make them more flexible and maintainable but I don't know when I'll have time.  I will at some point, though.

**Comment by apdavison** [comment]
Ready for review...

File: neo/core/basesignal.py [hunk 1]
```diff
@@ -139,8 +139,10 @@ def _get_required_attributes(self, signal, units):
         '''
         required_attributes = {}
         for attr in self._necessary_attrs:
-            if 'signal' == attr[0]:
-                required_attributes[str(attr[0])] = signal
+            if attr[0] == "signal":
+                required_attributes["signal"] = signal
+            elif attr[0] == "t_start":
+                required_attributes["t_start"] = getattr(self, attr[0], 0.0 * pq.ms)
```
Comment (samuelgarcia) [inline neo/core/basesignal.py:145]: required_attributes["t_start"] = getattr(self, "t_start", 0.0 * pq.ms)

**Comment by samuelgarcia** [comment]
This OK for me.

**Pull request merged**